sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/receipt/_claims_digest_computation.py

```diff
@@ -0,0 +1,192 @@
+            digest_claim = application_claim_object["digest"]
+            assert isinstance(digest_claim, dict)
+
+            # Assert on the digest value
+            assert "value" in digest_claim
+            assert isinstance(digest_claim["value"], str)
+
+            # Assert on the protocol
+            assert "protocol" in digest_claim
+            assert isinstance(digest_claim["protocol"], str)
+
+        else:
+            assert False, f"Unknown claim kind: {claim_kind}"
+
+
+def _compute_ledger_entry_v1_claim_digest(
+    ledger_entry_claim: LedgerEntryClaim,
+) -> bytes:
+    """Compute the digest of a LedgerEntryV1 claim. It returns the digest in bytes."""
+
+    # Decode the secret key
+    secret_key = b64decode(ledger_entry_claim.secretKey, validate=True)
+
+    # HMAC the collection ID with the secret key
+    collection_id_digest = hmac.new(
+        secret_key,
+        ledger_entry_claim.collectionId.encode(),
+        sha256,
+    ).digest()
+
+    # HMAC the ledger contents with the secret key
+    contents_digest = hmac.new(
+        secret_key,
+        ledger_entry_claim.contents.encode(),
+        sha256,
+    ).digest()
+
+    # Compute the SHA-256 of the concatenation of the collection ID and contents digests
+    return sha256(collection_id_digest + contents_digest).digest()
+
+
+def _compute_ledger_entry_claim_digest(ledger_entry_claim: LedgerEntryClaim) -> bytes:
+    """Compute the digest of a LedgerEntry claim. It returns the digest in bytes."""
+
+    claim_protocol = ledger_entry_claim.protocol
+
+    # Compute the digest based on the specified protocol
+    if claim_protocol == LEDGER_ENTRY_V1_CLAIM_PROTOCOL:
+        # Compute the digest of the LedgerEntryV1 claim
+        ledger_entry_digest = _compute_ledger_entry_v1_claim_digest(ledger_entry_claim)
+
+    else:
+        raise ValueError(f"Unsupported claim protocol: {claim_protocol}")
+
+    # Compute the SHA-256 of the concatenation of the protocol and the ledger entry digest
+    return sha256(claim_protocol.encode() + ledger_entry_digest).digest()
+
+
+def _compute_claim_digest_from_object(claim_digest_object: ClaimDigest) -> bytes:
+    # Compute the SHA-256 of the concatenation of the protocol and the digest value
+    return sha256(
+        claim_digest_object.protocol.encode() + bytes.fromhex(claim_digest_object.value)
+    ).digest()
+
+
+def _compute_claims_hexdigest(application_claims_list: List[ApplicationClaim]) -> str:
+    """Compute the CCF claims digest from the provided list of application claims objects.
+    It returns the hexdigest of the claims digest."""
+
+    # Initialize the claims digest
+    claims_digests_concatenation = b""
+
+    # Iterate through all the application claims objects to compute their single digest.
+    # We assume that the order of the application objects is valid
+    # and the digests will be concatenated in the same order.
+    for application_claim_object in application_claims_list:
```

Comment:
There looks to be three sets of iterations:
1. Validation
2. Conversion to object
3. Computing claims

We should try to get it done in one iteration as this is unnecessarily inefficient

Reply:
Yep, it is good that we react to syntax/formatting errors first if there is no hugely negative impact on performance. Totally forgot it is just one claim, so yeah performance is no big deal. Maybe add a comment explaining the reasoning + that there is not expected to be a performance hit.

Reply:
Sounds good, I added a comment to clarify this.